uisdk/src/main/java/com/karhoo/uisdk/util/CurrencyUtils.kt

```diff
@@ -1,58 +1,47 @@
 package com.karhoo.uisdk.util
 
+import java.math.BigDecimal
+import java.text.DecimalFormat
 import java.util.Currency
+import java.util.Locale
 
 object CurrencyUtils {
 
     fun getFormattedPrice(currency: String?, price: Int): String {
         return if (currency.isNullOrEmpty()) "" else {
             val currency = Currency.getInstance(currency)
-            intToPrice(currency, price)
+            currencyAwareFormatting(price.toLong(), currency)
         }
     }
 
     fun intToPrice(currency: Currency, price: Int): String {
-
-        val cost = price.toString()
-        val low = currency.defaultFractionDigits
-
-        val (costHi, costLow) = getDisplayPrice(cost, low)
-
-        return String.format("%s%s.%s", currency.symbol, costHi, costLow)
+        return currencyAwareFormatting(price.toLong(), currency)
     }
 
     fun intToRangedPrice(currency: Currency, lowPrice: Int, highPrice: Int): String {
-        val lowCost = lowPrice.toString()
-        val highCost = highPrice.toString()
-        val low = currency.defaultFractionDigits
-
-        val (costHiPrice, costHighFraction) = getDisplayPrice(highCost, low)
-        val (costLowPrice, costLowFraction) = getDisplayPrice(lowCost, low)
-
-        return String.format("%s%s.%s - %s.%s",
-                             currency.symbol,
-                             costLowPrice,
-                             costLowFraction,
-                             costHiPrice,
-                             costHighFraction)
+        val lowCostString = currencyAwareFormatting(lowPrice.toLong(), currency, includeCurrencySymbol = true)
+        val highCostString = currencyAwareFormatting(highPrice.toLong(), currency, includeCurrencySymbol = false)
+        return String.format("%s - %s", lowCostString, highCostString)
```

Comment:
Maybe add this in a constant

Reply:
Can you please explain what type of consistency are you referring to? Consistent with what?

Reply:
Since the common practice in the project seems to have extensions instead of utils, would it be better to have the CurrencyUtils transformed to a CurrencyExtension ?

Reply:
Done